modules/gapi/src/backends/fluid/gfluidcore.cpp

```diff
@@ -29,6 +29,24 @@
 #include <cstdlib>
 
 namespace cv {
+#if (CV_SIMD && !CV_NEON)
```

Comment:
CV_SIMD includes all types SIMD optimization such as: CV_SSE2, CV_AVX2, CV_AVX512 and CV_NEON. Since this SIMD opt don't give speed-up for ARM NEON, there are necessity to disable it for ARM NEON case.

Reply:
> SIMD opt don't give speed-up for ARM NEON

This looks really strange and unlikely.
Compilers just can't vectorize 8U integer arithmetic effectively from generic C++ code (due to C++ rules).

Do you have saved branch with enabled NEON?

Reply:
@alalek Yes it is very strange for me too. Orest tested this snippet on KMB with enabled NEON, but don't get speedup.

There is no branch with enabled NEON but I can create if it's necessary.

Reply:
@alalek @terfendail could you please advice another way to disable this snippet for the NEON case?

Reply:
If there are no noticeable regressions, then it make sense do not exclude NEON here.

Reply:
The point is that ARM is our target platform for optimization, so I can't reduce its performance even by a percentage.

Moreover, the decrease in performance is up to 15% on some test cases. Performance report is here:
[AddWeighted_neon_perf_report.xlsx](https://github.com/opencv/opencv/files/5886985/AddWeighted_neon_perf_report.xlsx)

Reply:
Reworked.

Reply:
Can't reproduce performance change observations on TX1 AARCH64 platform (Ubuntu 16.04 + GCC 5.4.0):

'PATCH" as commit 61e82884944c1e6d05c6bf30fa8c9ace483d35d2
"NO_SIMD" patch looks like this:
```.diff
-#if CV_SIMD
+#if 0
     x = addw_simd(in1, in2, out, _alpha, _beta, _gamma, length);
 #endif
```

Results for affected "1080" tests:

|Name of Test|NO_SIMD|PATCH|(x-factor)|
|---|:-:|:-:|:-:|
|TestPerformance::AddWeightedPerfTestFluid/AddWeightedPerfTest::(compare_f, 1920x1080, 8UC1, 0, { gapi.kernel_package })|13.698|9.106|1.50|
|TestPerformance::AddWeightedPerfTestFluid/AddWeightedPerfTest::(compare_f, 1920x1080, 8UC1, 5, { gapi.kernel_package })|3.504|3.490|1.00|
|TestPerformance::AddWeightedPerfTestFluid/AddWeightedPerfTest::(compare_f, 1920x1080, 8UC1, -1, { gapi.kernel_package })|13.715|9.091|1.51|
|TestPerformance::AddWeightedPerfTestFluid/AddWeightedPerfTest::(compare_f, 1920x1080, 16UC1, 0, { gapi.kernel_package })|15.818|5.398|2.93|
|TestPerformance::AddWeightedPerfTestFluid/AddWeightedPerfTest::(compare_f, 1920x1080, 16UC1, 5, { gapi.kernel_package })|3.726|3.744|1.00|
|TestPerformance::AddWeightedPerfTestFluid/AddWeightedPerfTest::(compare_f, 1920x1080, 16UC1, -1, { gapi.kernel_package })|16.571|5.423|3.06|
|TestPerformance::AddWeightedPerfTestFluid/AddWeightedPerfTest::(compare_f, 1920x1080, 16SC1, 0, { gapi.kernel_package })|15.776|5.380|2.93|
|TestPerformance::AddWeightedPerfTestFluid/AddWeightedPerfTest::(compare_f, 1920x1080, 16SC1, 5, { gapi.kernel_package })|3.739|3.741|1.00|
|TestPerformance::AddWeightedPerfTestFluid/AddWeightedPerfTest::(compare_f, 1920x1080, 16SC1, -1, { gapi.kernel_package })|18.858|5.429|3.47|
|TestPerformance::AddWeightedPerfTestFluid/AddWeightedPerfTest::(compare_f, 1920x1080, 8UC3, 0, { gapi.kernel_package })|40.477|26.670|1.52|
|TestPerformance::AddWeightedPerfTestFluid/AddWeightedPerfTest::(compare_f, 1920x1080, 8UC3, 5, { gapi.kernel_package })|9.827|9.823|1.00|
|TestPerformance::AddWeightedPerfTestFluid/AddWeightedPerfTest::(compare_f, 1920x1080, 8UC3, -1, { gapi.kernel_package })|40.471|26.656|1.52|
|TestPerformance::AddWeightedPerfTestFluid_short/AddWeightedPerfTest::(compare_f, 1920x1080, 16UC1, -1, { gapi.kernel_package })|16.534|5.439|3.04|
|TestPerformance::AddWeightedPerfTestFluid_short/AddWeightedPerfTest::(compare_f, 1920x1080, 16SC1, -1, { gapi.kernel_package })|18.934|5.442|3.48|

TX1 CPU is AARCH64-based ARM Cortex-A57 (quad-core) @ 1.73GHz
TX1 spec is here: https://elinux.org/Jetson_TX1

---

@terfendail Please try to collect results for this patch on the target platform.

Reply:
@alalek It seems to me that Vitaly is on vacation. So he can't collect results for this patch on the target platform. Could you please continue review without this step. Moreover, I have already provided the results for the target platform.